timemachine/cpp/src/context.cu

```diff
@@ -22,8 +22,16 @@ Context::Context(
     const double *box_0,
     std::shared_ptr<Integrator> intg,
     std::vector<std::shared_ptr<BoundPotential>> bps,
-    std::shared_ptr<MonteCarloBarostat> barostat)
-    : N_(N), barostat_(barostat), step_(0), d_sum_storage_(nullptr), d_sum_storage_bytes_(0), intg_(intg), bps_(bps) {
+    std::shared_ptr<MonteCarloBarostat> barostat,
+    bool freeze_reference)
```

Comment:
Chose this as a separate parameter so we could test it with and without, not really happy with it especially since if you freeze the reference right now you have to provide a temperature. I rely on the integrator for the temperature, though it reintroduces the issue that the selection idxs PR has fixed for @maxentile

Reply:
Is there a way to set context temperature if the integrator is VelocityVerlet?

Reply:
Not at the moment, just wasn't sure the best way to implement this.

Could have it so you can call `ensure_md_is_initialized` with a temperature which is honored if temperature > 0.0, otherwise tries to use the integrator's temperature.

Reply:
Reworked so that you can control local md behavior explicitly in https://github.com/proteneer/timemachine/pull/1014/commits/39087381d71a0f57369c6b08ac785e0442eb19b0